kamatera cloud provider unit tests are flaky

<!--
Please answer these questions before submitting your bug report. Thanks!
-->

**Which component are you using?**:

cluster-autoscaler

**What version of the component are you using?**:

HEAD commit from master

Component version:

**What k8s version are you using (`kubectl version`)?**:

n/a

**What environment is this in?**:

unit tests

**What did you expect to happen?**:

Tests should pass.

**What happened instead?**:

```
--- FAIL: TestNodeGroup_Others (0.01s)
    kamatera_node_group_test.go:285: 
        	Error Trace:	kamatera_node_group_test.go:285
        	Error:      	Not equal: 
        	            	expected: "node group ID: ng1 (min:1 max:7)\ninstance ID: 2773c4e7122d4bc58694ba4b275cfc0e state: Running powerOn: false\ninstance ID: 131a62b825a14f9c99335cd2d1e05ad5 state: Running powerOn: false\ninstance ID: bfccc7cc5ceb4011b32954c2d59186cb state: Running powerOn: false"
        	            	actual  : "node group ID: ng1 (min:1 max:7)\ninstance ID: 131a62b825a14f9c99335cd2d1e05ad5 state: Running powerOn: false\ninstance ID: bfccc7cc5ceb4011b32954c2d59186cb state: Running powerOn: false\ninstance ID: 2773c4e7122d4bc58694ba4b275cfc0e state: Running powerOn: false"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 node group ID: ng1 (min:1 max:7)
        	            	-instance ID: 2773c4e7122d4bc58694ba4b275cfc0e state: Running powerOn: false
        	            	 instance ID: 131a62b825a14f9c99335cd2d1e05ad5 state: Running powerOn: false
        	            	 instance ID: bfccc7cc5ceb4011b32954c2d59186cb state: Running powerOn: false
        	            	+instance ID: 2773c4e7122d4bc58694ba4b275cfc0e state: Running powerOn: false
        	Test:       	TestNodeGroup_Others
FAIL
FAIL	k8s.io/autoscaler/cluster-autoscaler/cloudprovider/kamatera	8.148s
```

**How to reproduce it (as minimally and precisely as possible)**:

Run tests.

**Anything else we need to know?**:

The test relies on specific order of map entries, which is not deterministic.

/assign @OriHoch

--- a/cluster-autoscaler/cloudprovider/kamatera/kamatera_node_group_test.go
+++ b/cluster-autoscaler/cloudprovider/kamatera/kamatera_node_group_test.go
@@ -21,6 +21,7 @@ import (
 	"fmt"
 	apiv1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
+	"strings"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
@@ -282,10 +283,12 @@ func TestNodeGroup_Others(t *testing.T) {
 	assert.Equal(t, 3, ts)
 	assert.Equal(t, "ng1", ng.Id())
 	assert.Equal(t, "node group ID: ng1 (min:1 max:7)", ng.Debug())
-	assert.Equal(t, fmt.Sprintf(`node group ID: ng1 (min:1 max:7)
-instance ID: %s state: Running powerOn: false
-instance ID: %s state: Running powerOn: false
-instance ID: %s state: Running powerOn: false`, serverName1, serverName2, serverName3), ng.extendedDebug())
+	extendedDebug := strings.Split(ng.extendedDebug(), "\n")
+	assert.Equal(t, 4, len(extendedDebug))
+	assert.Contains(t, extendedDebug, "node group ID: ng1 (min:1 max:7)")
+	for _, serverName := range []string{serverName1, serverName2, serverName3} {
+		assert.Contains(t, extendedDebug, fmt.Sprintf("instance ID: %s state: Running powerOn: false", serverName))
+	}
 	assert.Equal(t, true, ng.Exist())
 	assert.Equal(t, false, ng.Autoprovisioned())
 	_, err = ng.Create()

